### ISSUE TITLE
Not picking up workspaces from lerna.json

### ISSUE THREAD

**Issue by stepan662** [vercel/turborepo#688]
### What version of Turborepo are you using?

1.1.2

### What package manager are you using / does the bug impact?

npm

### What operating system are you using?

Mac

### Describe the Bug

Documentation mentions, that it's possible to use lerna.json for workspace definitions instead of package.json, however in my case this throws an error
```
could not detect workspaces: package.json: no workspaces found. Turborepo requires NPM workspaces to be defined in the root package.json
```

It now causes us issues with using npm for installation as npm is picking workspace definitions too, which we don't want.

### Expected Behavior

As I understand the documentation it should take packages definition from lerna.json

### To Reproduce

1. Create monorepo with turborepo cli `npx create-turbo@latest`
2. Install and intit lerna `npm install lerna && npx lerna init`
3. Move `workspaces` definition from package.json to lerna.json's `packages`
4. Run `npm run build`

**Comment by Xenfo** [comment]
It's not supposed to. Set `workspaces` key in your `package.json`.

**Comment by stepan662** [comment]
I've suspected so, however this part of the documentation is a bit confusing, maybe it would be worth to clarify how exactly it supports lerna:

> Turborepo considers Lerna a valid workspace implementation. Everything should work as expected.

We have an issue when using lerna and turborepo, when we need to define `workspaces` field into package json then it's picked up by npm when we install packages for the root and then tries to install packages for all workspaces (which we don't want as we are using `lerna bootstrap` command).

I've created a workaround by custom script which removes `workspaces` from package.json, then installs root packages and then restores package.json. However it's not very practical when you want to install new packages and so on. Do you have any advice how to solve this in a better way @Xenfo ? Lerna is optimal solution for us as we want the packages to be independent.

https://github.com/tolgee/tolgee-js/blob/main/package.json#L41

PS: Thanks for creating this project, it helps us a lot :)

**Comment by uglow** [comment]
TL;DR: Can you be explicit concerning how to use Turborepo with Lerna when workspaces are not involved, please?

@Xenfo @jaredpalmer - the [documentation](https://turborepo.org/docs/guides/migrate-from-lerna) says:

> If you're not using yarn or NPM workspaces with Lerna, you should keep around your `lerna bootstrap` command if you have one. Turborepo considers Lerna a valid workspace implementation. Everything should work as expected.

I thought I'll try out Turborepo like @stepan662 , but got to the same point. I don't want to start messing around with NPM workspaces as I'm using Lerna already. From my reading of the docs, I took it to mean, "If you're not using yarn or NPM workspaces with Lerna...Turborepo considers Lerna a valid workspace implementation. Everything should work as expected.". Clearly that's not correct.

What does "you should keep around your `lerna bootstrap` command if you have one" mean, exactly?

**Comment by edgaryp** [comment]
> I've created a workaround by custom script which removes `workspaces` from package.json, then installs root packages and then restores package.json. However it's not very practical when you want to install new packages and so on.

@stepan662 I am also running into the same issue and discovered that `--workspaces` can be disabled like this `npm install --workspaces=false` per the official [doc](https://docs.npmjs.com/cli/v8/commands/npm-install).

**Comment by stepan662** [comment]
@edgaryp Hmm, strangely this doesn't work for me. Also I didn't find this in the docs anywhere. However even if it works it's not ideal as you would need to add this parameter everywhere (e.g. when you install new package in the root).

**Comment by omeraplak** [comment]
Hey,
Thank you for your great effort. We have the same issue on [refine repo](https://github.com/pankod/refine/pull/1607). Thanks a lot @stepan662 for bash script but I hate bash scripts :) 

Will this be fixed in the future? We have 55+ examples and we are running them conditionally (with scopes). Doing `npm i` is an endless adventure for around 70+ projects. Thank you for your time!

**Comment by snapwich** [comment]
same issue here. adding workspaces to package.json causes `npm install` to attempt installing all packages across repos which is not wanted. also, for whatever reason and despite what the documentation says, `npm install --workspaces=false` behaves the same as `npm install`. this is on npm version `8.1.0`

**Comment by JonRobinson-Proximie** [comment]
Same problem here - since Lerna is now under Nrwl's stewardship per https://github.com/lerna/lerna/issues/3121 and https://blog.nrwl.io/lerna-is-dead-long-live-lerna-61259f97dbd9 I'm looking in that direction

**Comment by ghiscoding** [comment]
Here's a possible long explanation to try to ease the understanding of the `workspaces` vs `packages` in Lerna, at least this is how I understand it... 

Lerna needs to know where your monorepo packages are (it needs to know that if you use the `bootstrap` command but also when scoping commands to certain packages and most importantly it also needs to know the location so that it can build the package graph topology, that is which package(s) are dependent on which, etc...).

At some point, Lerna added a flag named `--use-workspaces` and it was added especially for Yarn (at the time) when they came up with [Yarn Workspaces](https://classic.yarnpkg.com/en/docs/workspaces) and they decided to add a `workspaces` property in the project root `package.json` (hence the name of Lerna's flag, "use-workspaces"). Do you have to use it? No, you can still use `packages` in `lerna.json` and that would still work but you'll then have 2 places to define your workspace packages and that would work too but is less ideal, but again both ways would work (Lerna will use "workspaces" from `package.json` only if  `--use-workspaces` is enabled or else it will use "packages" from `lerna.json`). So the `--use-workspaces` is just a short way of saying, "hello from Yarn, we defined all workspace packages in `package.json`, we might as well use that list for Lerna too". Also note that even though the flag `--use-workspaces` was added for Yarn, it is now also working with npm 7+ workspaces because they also chose to define their workspace packages through the same "workspaces" property in `package.json` as shown in their [docs](https://docs.npmjs.com/cli/v8/using-npm/workspaces#defining-workspaces))

However, the `--use-workspaces` flag does **not** work with pnpm (or at least it won't do anything with it and might cause confusion) because they do not use, neither read "workspaces" from `package.json` because they decided to use a separate `pnpm-workspace.yaml` file (more info at [pnpm-workspace.yaml](https://pnpm.io/pnpm-workspace_yaml)) and for that reason `--use-workspaces` is useless with pnpm and it would be preferable to stick with "packages" defined in `lerna.json`

I hope this might clears some questions or misunderstandings about the `--use-workspaces` flag from Lerna. The short summary is you can enable `--use-workspaces` if supported but if it's causing trouble with your package manager then just keep using "packages" defined in `lerna.json`

Another thing to note, Lerna `bootstrap` command should be deprecated (see this [comment](https://github.com/lerna/lerna/discussions/3140#discussioncomment-3033079) by Lerna's author), because all package manager clients now support Workspaces and they do a much better job at symlinking packages than Lerna could ever do... so you should stop using Lerna `bootstrap`

Lastly [Lerna-Lite](https://github.com/ghiscoding/lerna-lite), can be an alternative to Lerna and it works great with Turborepo... it's smaller and has couple of extra features like the support of the `workspace:` protocol from [pnpm `workspace:`](https://pnpm.io/workspaces#workspace-protocol-workspace) or [yarn `workspace:`](https://yarnpkg.com/features/workspaces#workspace-ranges-workspace) which is not currently supported by Lerna (see [issue](https://github.com/lerna/lerna/pull/2450)).

sorry for the long post, hope it helps. Cheers and happy coding.

**Comment by c4nzin** [comment]
> It's not supposed to. Set `workspaces` key in your `package.json`.

thank you !
